toothpick-runtime/src/main/java/toothpick/ScopeImpl.java

```diff
@@ -161,22 +161,22 @@ private void installModule(Module module) {
   //make the APIs very unstable as you could not get any instance of the
   //implementation class via an scope, it would fail but be syntactically valid.
   //only creating an instance of the interface is valid with this syntax.
-  /*VisibleForTesting*/ <T> ScopedProviderImpl<T> toProvider(Binding<T> binding) {
+  /*VisibleForTesting*/ <T> InternalProviderImpl<T> toProvider(Binding<T> binding) {
     if (binding == null) {
       throw new IllegalStateException("null binding are not allowed. Should not happen unless getBindingSet is overridden.");
     }
     switch (binding.getMode()) {
       case SIMPLE:
-        return new ScopedProviderImpl<>(this, binding.getKey(), false);
+        return new InternalProviderImpl<>(binding.getKey(), false);
       case CLASS:
-        return new ScopedProviderImpl<>(this, binding.getImplementationClass(), false);
+        return new InternalProviderImpl<>(binding.getImplementationClass(), false);
       case INSTANCE:
-        return new ScopedProviderImpl<>(binding.getInstance());
+        return new InternalProviderImpl<>(binding.getInstance());
       case PROVIDER_INSTANCE:
         //to ensure providers do not have to deal with concurrency, we wrap them in a thread safe provider
-        return new ScopedProviderImpl<>(binding.getProviderInstance(), false);
+        return new InternalProviderImpl<>(binding.getProviderInstance(), false);
       case PROVIDER_CLASS:
-        return new ScopedProviderImpl<>(this, binding.getProviderClass(), true);
+        return new InternalProviderImpl<>(binding.getProviderClass(), true);
```

Comment:
I wonder if the only problem was not here. The providers created out of bindings are indeed not scoped.